cpp/tests/geometry/TriangleMesh.cpp

```diff
@@ -1843,33 +1843,121 @@ TEST(TriangleMesh, CreateFromPointCloudPoisson) {
-            {6, 11, 34},  {34, 29, 6},  {12, 10, 33}, {33, 35, 12},
-            {34, 12, 35}, {11, 12, 34}, {24, 23, 48}, {36, 23, 25},
-            {36, 37, 23}, {37, 48, 23}, {38, 24, 48}, {38, 39, 24},
-            {39, 26, 24}, {39, 49, 26}, {38, 48, 37}, {25, 27, 40},
-            {40, 36, 25}, {27, 28, 41}, {41, 40, 27}, {28, 29, 42},
-            {42, 41, 28}, {39, 30, 49}, {39, 31, 30}, {39, 50, 31},
-            {39, 43, 50}, {44, 50, 43}, {32, 31, 50}, {44, 32, 50},
-            {44, 45, 32}, {45, 33, 32}, {42, 34, 46}, {29, 34, 42},
-            {47, 33, 45}, {35, 33, 47}, {34, 35, 47}, {47, 46, 34},
-            {37, 36, 51}, {39, 38, 52}, {53, 52, 51}, {52, 37, 51},
-            {52, 38, 37}, {36, 40, 54}, {54, 51, 36}, {54, 40, 41},
-            {51, 54, 55}, {55, 53, 51}, {55, 41, 42}, {54, 41, 55},
-            {43, 39, 52}, {56, 52, 53}, {56, 44, 52}, {44, 43, 52},
-            {45, 44, 56}, {56, 55, 57}, {53, 55, 56}, {55, 42, 46},
-            {46, 57, 55}, {45, 56, 57}, {57, 47, 45}, {57, 46, 47}};
+        {1, 13, 0},
+        {0, 14, 2},
+        {13, 14, 0},
+        {1, 15, 13},
+        {15, 16, 13},
+        {1, 3, 15},
+        {14, 13, 16},
+        {16, 17, 14},
+        {2, 18, 4},
+        {14, 18, 2},
+        {4, 18, 5},
+        {18, 14, 17},
+        {17, 19, 18},
+        {18, 19, 6},
+        {6, 5, 18},
+        {7, 16, 15},
+        {7, 8, 16},
+        {8, 20, 16},
+        {21, 16, 20},
+        {17, 16, 21},
+        {9, 20, 8},
+        {10, 20, 9},
+        {21, 20, 10},
+        {22, 17, 21},
+        {19, 17, 22},
+        {19, 22, 11},
+        {11, 6, 19},
+        {22, 21, 10},
+        {10, 12, 22},
+        {11, 22, 12},
+        {24, 0, 23},
+        {1, 0, 24},
+        {0, 2, 25},
+        {25, 23, 0},
+        {3, 1, 24},
+        {24, 26, 3},
+        {2, 4, 27},
+        {27, 25, 2},
+        {27, 5, 28},
+        {4, 5, 27},
+        {28, 6, 29},
+        {5, 6, 28},
+        {8, 7, 30},
+        {30, 31, 8},
+        {32, 8, 31},
+        {9, 8, 32},
+        {33, 9, 32},
+        {10, 9, 33},
+        {6, 11, 34},
+        {34, 29, 6},
+        {12, 10, 33},
+        {33, 35, 12},
+        {34, 12, 35},
+        {11, 12, 34},
+        {24, 23, 48},
+        {36, 23, 25},
+        {36, 37, 23},
+        {37, 48, 23},
+        {38, 24, 48},
+        {38, 39, 24},
+        {39, 26, 24},
+        {39, 49, 26},
+        {38, 48, 37},
+        {25, 27, 40},
+        {40, 36, 25},
+        {27, 28, 41},
+        {41, 40, 27},
+        {28, 29, 42},
+        {42, 41, 28},
+        {39, 30, 49},
+        {39, 31, 30},
+        {39, 50, 31},
+        {39, 43, 50},
+        {44, 50, 43},
+        {32, 31, 50},
+#if defined(__APPLE__) && defined(__arm64__)
+        // Unkown why these two triangles are different on Apple Silicon
+        {44, 45, 50},
+        {45, 32, 50},
+#else
+        {44, 32, 50},
+        {44, 45, 32},
+#endif
```

Comment:
Looks like the set of 4 vertices is triangulated in two different but equivalent ways. Numerical differences can cause this. Much more common when you compare CPU vs CUDA results.

Reply:
I was thinking along those lines, I've updated the code comment to a clearer version